www/_include.php

```diff
@@ -96,13 +96,23 @@ function __toString() {
 /* Set the $SIMPLESAML_INCPREFIX to a reference to the class. */
 $SIMPLESAML_INCPREFIX = new SimpleSAML_IncPrefixWarn();
 
+/* Get environment variable that specifies an alernate location for the configuration files. */
+$configdir = getenv("SIMPLESAMLPHP_CONFIG_DIR");
 
-$configdir = dirname(dirname(__FILE__)) . '/config';
-if (!file_exists($configdir . '/config.php')) {
-	header('Content-Type: text/plain');
-	echo("You have not yet created the simpleSAMLphp configuration files.\n");
-	echo("See: http://rnd.feide.no/content/installing-simplesamlphp#id434777\n");
-	exit(1);
+/* Check if environment variable is set and if the directory exists.
+ * If neither condition is met use the default configuration directory.
+ */
+if (empty($configdir) || !file_exists($configdir)) {
+	$configdir = dirname(dirname(__FILE__)) . '/config';
+	if (!file_exists($configdir . '/config.php')) {
+		header('Content-Type: text/plain');
+		echo("You have not yet created the simpleSAMLphp configuration files.\n");
+		echo("See: http://rnd.feide.no/content/installing-simplesamlphp#id434777\n");
+		exit(1);
+	}
+} else {
+	/* Set the configuration directory to the new location. */
+	SimpleSAML_Configuration::setConfigDir($configdir);
```

Comment:
And here you are assuming that if someone specified a directory (well, or a file actually), it will be a valid configuration directory containing the _config.php_ file. We should not assume that and explicitly check it in the same way we do when we use a pre-fixed location.